src/components/elements/tableFilters/filters/ColumnsMenu.tsx

```diff
@@ -36,6 +36,12 @@ const TableColumnsMenu = ({
         setColumnsValue(defaultValue);
         reset();
       }}
+      // When the popover opens, set the state to whatever is currently in columnsValue.
+      // columnsValue may change on renders subsequent to the first one (see useOptionalColumns hook),
+      // so this menu component needs to "catch up" in order to display the up-to-date state!
+      onOpen={() => setState(columnsValue)}
```

Comment:
Here's a repro of the problem this is trying to solve:
- comment out this line (remove my workaround)
- go to a table and select some optional columns
- navigate away to another page
- navigate back, not using the back button, but using in-app links
- the table will load correctly, and the search params will update to reflect your saved preference for optional columns, BUT the columns popover will not show any columns selected:
![Screenshot 2025-02-25 at 4 01 13 PM](https://github.com/user-attachments/assets/01695ed7-02e7-4614-916b-fb2576f82dce)

Reply:
Cool this can now be removed :)